### PR TITLE
feat: support legacy config

### LINKED ISSUE #5: Not working in non-flat config usage
Hey team,

Looks like the plugin is not adding nor removing any rules on an empty configuration.

Repro:

```bash
npx create-react-app eslint-repro
cd eslint-repro
npm init @eslint/config
npx eslint --print-config src/App.js # check output
npm i eslint-plugin-oxlint
# here, add it to the eslint config's plugin list, then
npx eslint --print-config src/App.js  # same output (some rules should be off'd)
```

We might be missing a couple test to ensure final eslint config has been modified.

edit: looks like I'm not using eslint 9.0.0 defaults yet, and it might be related.
edit2: looks like rules are the same ib both configuration formats ...: https://eslint.org/docs/latest/extend/plugin-migration-flat-config#migrating-rules-for-flat-config

### PR DESCRIPTION
**Pull request by Djiit**
Fixes #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,21 +10,43 @@ You can see https://oxc-project.github.io/blog/2023-12-12-announcing-oxlint.html
...
+If you are using legacy configuration (eslint < 9.0), you can use the following config:
+
+```js
+// .eslintrc.js
+const { oxlintRules } = require('eslint-plugin-oxlint')
+module.exports = {
+  ... // other config
+  rules: {
+    ... // other rules
+    ...oxlintRules // oxlint rules should be the last ones
```
Comment (Dunqing) [inline README.md:43]: I think we should support passing ' plugins: [oxlint]` when the eslint version < 9.0. But I haven't looked into what needs to be done.

**Review by Dunqing — CHANGES_REQUESTED** [review]
Thank you for improving the documentation!!

File: README.md (hunk 1 above)
Comment (Djiit) [reply README.md:43]: I agree with you. In the meantime, doing like this works well and is not too hacky to setup, imho.

**Comment by Djiit** [comment]
Using conditional exports, we should be able to do something. See: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1886

**Comment by Djiit** [comment]
@Dunqing alright, if figured it out. The "presets" in eslint legacy config are called "configs" in the docs... See: https://eslint.org/docs/latest/extend/plugins#configs-in-plugins

Now, the plugin can be used with one line of config in legacy configs :)

**Comment by Djiit** [comment]
I noticed the repo don't have any formatter configured -- looks like my formatter (prettier) did some changes. Can be reverted on demand.

**Comment by Dunqing** [comment]
> @Dunqing alright, if figured it out. The "presets" in eslint legacy config are called "configs" in the docs... See: https://eslint.org/docs/latest/extend/plugins#configs-in-plugins
> 
> Now, the plugin can be used with one line of config in legacy configs :)

Thanks! That's great!

**Review by Dunqing — APPROVED** [review]
Thanks for working on this!

**Pull request merged**